### ISSUE TITLE
Support for linear scheduler with warmup (finetune like BERT does)

### ISSUE THREAD

**Issue by ciaochiaociao** [flairNLP/flair#2396]
Hello,

I would like to run some experiments based on transformer-like architecture, e.g., BERT + Linear + CRF, and **finetune** the BERT layer. 

The common practice for finetune it is using AdamW as optimizer and a learning rate scheduler with warmup at first and linear decay afterwards.

I know in trainer, it can customize optimizer like

```
from torch.optim import AdamW

trainer = ModelTrainer(...,optimizer=AdamW)  # takes as input a class

trainer.train(...)
```

But when it comes to customize **scheduler**, things get weird. For example, using huggingface's `transformers` package to get a linear scheduler with warmup:

```
from transformers import get_linear_schedule_with_warmup
scheduler = get_linear_schedule_with_warmup(optimizer=AdamW(...))  # optimizer takes as input an instance and this function returns a scheduler INSTANCE
```

Note one has to **instantiate** an optimizer beforehand.

And then to customize scheduler during training
```
trainer.train(scheduler=scheduler)  # this seems not to work for it only consumes a class instead of an instance
```

So to make this work, maybe we can make 
 1. `ModelTrainer` accept **optimizer object** AND `trainer.train()` accept **scheduler object** OR
 2. Create a `LinearSchedulerWithWarmup` class, so it can be input to `.train()`

Do you plan to do this or have any suggestions?

**Comment by alanakbik** [comment]
@ciaochiaociao good point and something we should really add to Flair! @lukasgarbas can you take a look at this?
